bofire/kernels/categorical.py

```diff
@@ -0,0 +1,25 @@
+import torch
+from gpytorch.kernels.kernel import Kernel
+from torch import Tensor
+
+
+class HammingKernelWithOneHots(Kernel):
+    has_lengthscale = True
+
+    def forward(
```

Comment:
Are you sure that this actually the same as the botorch implementation. From my understanding you are introducing here one lenghtscale per one-hotted column, whereas as the botorch had one lengthscale per feature. In the end, this appears to me to be the same as a RBF kernel under the assumption that one always has a zero or one, or?

For me the step is missing here where we go from one-hot back to ordinal encoding ..., but I could be wrong.